fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/auto/DataSourceConfiguration.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.wiki.webservice.data.config.auto;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.data.time.TimeGrain;
+
+import java.util.List;
+
+/**
+ * Holds the minimum necessary configuration necessary to set up fili to
+ * make requests to druid. This defines all metrics, dimensions, and valid
+ * time grains of a datasource.
+ */
+public interface DataSourceConfiguration {
+    /**
+     * Gets the name of a datasource as would be stored in Druid.
+     * @return the name of the datasource.
+     */
+    String getName();
+
+    /**
+     * Gets the name of the datasource to be used as a {@link TableName} in fili.
+     * @return the {@link TableName} for this datasource.
+     */
+    TableName getTableName();
+
+    /**
+     * Gets the names of all the metrics for the current datasource.
+     * @return a list of names of metrics for the current datasource.
+     */
+    List<String> getMetrics();
+
+    /**
+     * Gets the names of all the dimensions for the current datasource.
+     * @return a list of names of dimensions for the current datasource.
+     */
+    List<String> getDimensions();
+
+    /**
+     * Gets the list of {@link TimeGrain} which are valid for use in queries.
+     * @return a list of {@link TimeGrain} for the current datasource.
+     */
+    List<TimeGrain> getValidTimeGrains();
```

Comment:
I feel like this method is trying to be a little bit too smart... If this class is modeling a Druid DataSource (and it sure feels like that's what it's doing), then it should model just a single time grain, since a Druid data source _should_ have just a single grain. 

The other smarts that this method is wrapping should be someone else's responsibilities.

In theory, a Datasource in Druid _could_ have more than 1 grain at different times, but for now we only want to support one. It would be worth calling out this restriction somewhere as well (possibly in the README as well as the JavaDoc on this class)